### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by AjayThorve**
Update docs as per the offline discussion with @exactlyallan.

### CODE REVIEW

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
actually, I'd like to request changes on the `update-version.sh` script. blocking the merge while I type up a proper review.

File: README.md [hunk 1]
```diff
@@ -171,6 +171,18 @@ conda install -c rapidsai-nightly -c numba -c conda-forge -c nvidia \
 
 Note: cuxfilter is supported only on Linux, and with Python versions 3.8 and later.
 
+### Pypi
+
+Install cuxfilter from pypi using pip:
```
Comment (ajschmidt8) [inline README.md:176]:
```suggestion
### PyPI

Install cuxfilter from PyPI using pip:
```

this seems like the stylistic convention used by PyPI.

File: ci/release/update-version.sh [hunk 2]
```diff
@@ -36,6 +36,8 @@ function sed_runner() {
 # RTD update
 sed_runner 's/version = .*/version = '"'${NEXT_SHORT_TAG}'"'/g' docs/source/conf.py
 sed_runner 's/release = .*/release = '"'${NEXT_FULL_TAG}'"'/g' docs/source/conf.py
+# docs update
+sed_runner "s/cuxfilter=${CURRENT_SHORT_TAG}/cuxfilter=${NEXT_SHORT_TAG}/g" docs/source/user_guide/installation.rst
```
Comment (ajschmidt8) [inline ci/release/update-version.sh:40]:
```suggestion
sed_runner "/cuxfilter=[0-9]\{2\}.[0-9]\{2\}/ s/=[0-9]\{2\}.[0-9]\{2\}/=${NEXT_SHORT_TAG}/g" docs/source/user_guide/installation.rst
```

For the reasons outlined in the comment below, I discourage the use of variables in the search pattern of the `sed` expression.

- https://github.com/rapidsai/ucxx/pull/36#discussion_r1178429239

File: docs/source/user_guide/installation.rst [hunk 3]
```diff
@@ -5,23 +5,39 @@ Conda
 -----
 For the most customized way of installing RAPIDS and cuxfilter, visit the selector on the RAPIDS `Install Page <https://docs.rapids.ai/install>`_.
 
-cuxfilter conda example installation:
-
 .. code-block:: bash
 
+    # for CUDA 12.0
+    conda install -c rapidsai -c conda-forge -c nvidia \
+        cuxfilter=23.12 python=3.10 cuda-version=12.0
```
Comment (ajschmidt8) [inline docs/source/user_guide/installation.rst:12]:
```suggestion
        cuxfilter=23.10 python=3.10 cuda-version=12.0
```

These should be `23.10`, right? `23.12` doesn't exist yet.

File: docs/source/user_guide/installation.rst [hunk 4]
```diff
@@ -5,23 +5,39 @@ Conda
...
+
+    # for CUDA 11.8
     conda install -c rapidsai -c conda-forge -c nvidia \
-        cuxfilter=23.6 python=3.10 cudatoolkit=11.8
+        cuxfilter=23.12 python=3.10 cuda-version=11.8
```
Comment (ajschmidt8) [inline docs/source/user_guide/installation.rst:16]:
```suggestion
        cuxfilter=23.10 python=3.10 cuda-version=11.8
```

These should be `23.10`, right? `23.12` doesn't exist yet.

File: docs/source/user_guide/installation.rst [hunk 5]
```diff
@@ -5,23 +5,39 @@ Conda
...
+
+Pypi
+----
+Install cuxfilter from pypi using pip:
```
Comment (ajschmidt8) [inline docs/source/user_guide/installation.rst:20]:
```suggestion
PyPI
----
Install cuxfilter from PyPI using pip:
```

Minor stylistic suggestion

File: ci/release/update-version.sh [hunk 6]
```diff
@@ -53,11 +55,11 @@ for FILE in dependencies.yaml conda/environments/*.yaml; do
   done
 done
 
-
 # README.md update
 sed_runner "s/version == ${CURRENT_SHORT_TAG}/version == ${NEXT_SHORT_TAG}/g" README.md
```
Comment (ajschmidt8) [inline ci/release/update-version.sh:59]:
```suggestion
sed_runner "/version == / s/== .*\`/== ${NEXT_SHORT_TAG}\`/g" README.md
```

Suggesting the removal of the variable in the search pattern.

File: ci/release/update-version.sh [hunk 7]
```diff
@@ -53,11 +55,11 @@ for FILE in dependencies.yaml conda/environments/*.yaml; do
...
 sed_runner "s/cuxfilter=${CURRENT_SHORT_TAG}/cuxfilter=${NEXT_SHORT_TAG}/g" README.md
```
Comment (ajschmidt8) [inline ci/release/update-version.sh:60]:
```suggestion
sed_runner "/cuxfilter=[0-9]\{2\}.[0-9]\{2\}/ s/=[0-9]\{2\}.[0-9]\{2\}/=${NEXT_SHORT_TAG}/g" README.md
```

Suggesting the removal of the variable in the search pattern.

File: ci/release/update-version.sh
Comment (ajschmidt8) [inline ci/release/update-version.sh:1]: GitHub won't let me make this suggestion in the PR review, but can you change line 52 inside the `for` loops to:

```sh
sed_runner "/-.* ${DEP}==/ s/==.*/==${NEXT_SHORT_TAG_PEP440}.*/g" ${FILE}
```

This minor change ensures that the search pattern will work on lines in `dependencies.yaml` that use YAML anchors.

It matches what we've done for `cudf` and some other repositories: https://github.com/rapidsai/cudf/blob/5eee8ac988686dd8e0cc8328194055aa7579d9b6/ci/release/update-version.sh#L96C5-L96C78

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]

File: ci/release/update-version.sh
Comment (AjayThorve) [reply ci/release/update-version.sh:1]: applied the suggested changes

**Review by ajschmidt8 — APPROVED** [review]

**Review by exactlyallan — APPROVED** [review]
Looks good, the extra explanation around graphs is very helpful.

**Pull request merged**